contracts/Committee.sol

```diff
@@ -5,79 +5,224 @@ import "./spec_interfaces/IValidatorsRegistration.sol";
+			weight: data.weight
+		});
+		data.inCommittee = true;
+		data.pos = entryPos;
+		info.committeeBitmap |= uint32(uint(1) << entryPos);
+		info.committeeSize++;
+
+		sortBytes[info.committeeSize - 1] = byte(entryPos);
+		return (info, sortBytes);
+	}
+
+	function removeMemberFromCommittee(MemberData memory data, bytes memory sortBytes, CommitteeInfo memory info) private returns (CommitteeInfo memory newInfo, bytes memory newSortBytes) {
+		uint rank = 0;
+		while (uint8(sortBytes[rank]) != data.pos) {
+			rank++;
+		}
+
+		for (; rank < info.committeeSize - 1; rank++) {
+			sortBytes[rank] = sortBytes[rank + 1];
+		}
+		sortBytes[rank] = 0;
+
+		info.committeeSize--;
+		if (info.committeeSize > 0) {
+			info.minCommitteeMemberPos = uint8(sortBytes[info.committeeSize - 1]);
+		}
+		info.committeeBitmap &= ~uint32(uint(1) << data.pos);
+
+		delete committee[data.pos];
+		data.inCommittee = false;
+
+		return (info, sortBytes);
+	}
+
+	function removeMemberAtPos(uint8 pos, bytes memory sortBytes, CommitteeInfo memory info) private returns (CommitteeInfo memory newInfo, bytes memory newSortBytes) {
+		if (info.committeeBitmap & (uint(1) << pos) == 0) {
+			return (info, sortBytes);
+		}
+
+		address addr = committee[pos].addr;
+		MemberData memory data = membersData[addr];
+
+		(newInfo, newSortBytes) = removeMemberFromCommittee(data, sortBytes, info);
+
+		emit ValidatorCommitteeChange(addr, data.weight, data.isCompliant, false);
+
+		membersData[addr] = data;
+	}
+
+	function rankMember(address addr, MemberData memory data, bytes memory sortBytes, CommitteeInfo memory info) private view returns (CommitteeInfo memory newInfo, bytes memory newSortBytes) {
+		uint rank = 0;
+		while (uint8(sortBytes[rank]) != data.pos) {
+			rank++;
+		}
+
+		CommitteeMember memory cur = CommitteeMember({addr: addr, weight: data.weight});
+		CommitteeMember memory next;
+
+		while (rank < info.committeeSize - 1) {
+			next = committee[uint8(sortBytes[rank + 1])];
+			if (cur.weight > next.weight || cur.weight == next.weight && cur.addr > next.addr) break;
+
+			(sortBytes[rank], sortBytes[rank + 1]) = (sortBytes[rank + 1], sortBytes[rank]);
+			rank++;
+		}
+
+		while (rank > 0) {
+			next = committee[uint8(sortBytes[rank - 1])];
+			if (cur.weight < next.weight || cur.weight == next.weight && cur.addr < next.addr) break;
+
+			(sortBytes[rank], sortBytes[rank - 1]) = (sortBytes[rank - 1], sortBytes[rank]);
+			rank--;
+		}
+
+		info.minCommitteeMemberPos = uint8(sortBytes[info.committeeSize - 1]);
+		return (info, sortBytes);
+	}
+
+	function getMinCommitteeMemberWeight() external view returns (uint96) {
+		return committee[committeeInfo.minCommitteeMemberPos].weight;
+	}
+
+	function assignRewardsIfNeeded(Settings memory _settings) private {
+        IRewards rewardsContract = getRewardsContract();
+        uint lastAssignment = rewardsContract.getLastRewardAssignmentTime();
+        if (now - lastAssignment < _settings.maxTimeBetweenRewardAssignments) {
```

Comment:
<= to avoid multiple checks in the same transaction, for example on delegation change

Reply:
This is only relevant for the case of `maxTimeBetweenRewardAssignments == 0` right?

Reply:
right